src/radical/pilot/agent/launch_method/srun.py

```diff
@@ -152,12 +161,11 @@ def get_launch_cmds(self, task, exec_path):
         if self._rm_info.get('gpus'):
             mapping += ' --gpus-per-task %d' % n_gpus
 
-        if self._vmajor > 18:
-            if nodefile:
-                mapping += ' --nodefile=%s' % nodefile
-        else:
-            if nodelist:
-                mapping += ' --nodelist=%s' % ','.join(str(n) for n in nodelist)
+        if nodefile:
+            mapping += ' --nodefile=%s' % nodefile
+
+        if nodelist:
```

Comment:
`elif`?  We should never have both.  In fact, I would say it is 
```
if nodefile:
    ...
else:
    ...
```

Reply:
We would never have both @andre-merzky , as you can see if we have `nodfile` then once we finish writing to that `nodefile` from the `nodelist` we clear the `nodelist`. 

```python
with ru.ru_open(nodefile, 'w') as four:
        fout.write(','.join(nodelist))
        fout.write('\n')
nodelist.clear()